packages/backend/src/app/workers/flow-worker/queues/redis/redis-queue.ts

```diff
@@ -31,6 +39,63 @@ let scheduledJobQueue: Queue<ScheduledJobData, unknown>
 
 const repeatingJobKey = (id: ApId): string => `activepieces:repeatJobKey:${id}`
 
+const queueEnabled = system.getBoolean(SystemProp.QUEUE_UI_ENABLED)
+const queueUsername = system.get(SystemProp.QUEUE_UI_USERNAME)
+const queuePassword = system.get(SystemProp.QUEUE_UI_PASSWORD)
+const edition = getEdition()
+const QUEUE_BASE_PATH = '/ui'
+
+export async function setupBullMQBoard(app: FastifyInstance): Promise<void> {
+    logger.info(system.get(SystemProp.QUEUE_UI_ENABLED) + ' ' + queueEnabled, '[setupBullMQBoard] queueEnabled')
+    if (!queueEnabled) {
+        return
+    }
+    if (edition === ApEdition.CLOUD) {
+        return
+    }
+    logger.info('[setupBullMQBoard] Setting up bull board, visit /ui to see the queues')
+
+    await app.register(basicAuth, {
+        validate: (username, password, _req, reply, done) => {
+            if (username === queueUsername && password === queuePassword) {
+                done()
+            }
+            else {
+                done(new Error('Unauthorized'))
+            }
+        },
+        authenticate: true,
+    })
+
+
+    const serverAdapter = new FastifyAdapter()
+    createBullBoard({
+        queues: [new BullMQAdapter(oneTimeJobQueue), new BullMQAdapter(scheduledJobQueue)],
+        serverAdapter,
+    })
+    serverAdapter.setBasePath(QUEUE_BASE_PATH)
+
+    app.addHook('onRequest', (req, reply, next) => {
+        if (!req.routerPath.startsWith(QUEUE_BASE_PATH)) {
+            next()
+        } 
+        else {
+            app.basicAuth(req, reply, function (error) {
```

Comment:
The plugin doesn't seem to support async mode, @khaledmashaly double check if you can